strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/AuthorizationConfigController.java

```diff
@@ -147,88 +140,206 @@ public ResponseEntity getAuthorizationConfig()
-                                                                   .filter(
-                                                                           role1 -> role1.getName()
-                                                                                         .equalsIgnoreCase(
-                                                                                                 role.getName()))
-                                                                   .forEach(
-                                                                           foundedRole -> foundedRole.getPrivileges()
-                                                                                                     .forEach(
-                                                                                                             this::addAnonymousAuthority))));
+        if (roleListForm == null)
+        {
+            throw new RequestBodyValidationException("Empty request body", bindingResult);
+        }
+        if (bindingResult.hasErrors())
+        {
+            String message = "Roles cannot be saved because the submitted form contains errors!";
+            throw new RequestBodyValidationException(message, bindingResult);
+        }
+
+        // Convert RoleListForm to List<Role>
+        TypeDescriptor sourceType = TypeDescriptor.valueOf(RoleListForm.class);
+        TypeDescriptor targetType = TypeDescriptor.collection(List.class, TypeDescriptor.valueOf(Role.class));
+        List<Role> roleList = (List<Role>) conversionService.convert(roleListForm,
+                                                                     sourceType,
+                                                                     targetType);
+
+        return processConfig(config -> addRolesToAnonymous(config, Objects.requireNonNull(roleList)),
+                             () -> SUCCESSFUL_ASSIGN_ROLES,
+                             acceptHeader);
+    }
+
+    private void addRolesToAnonymous(AuthorizationConfig config,
+                                     List<Role> roles)
+    {
+        roles.forEach(role -> config.getRoles()
+                                    .getRoles()
+                                    .stream()
+                                    .filter(
+                                            role1 -> role1.getName()
+                                                          .equalsIgnoreCase(
+                                                                  role.getName()))
+                                    .forEach(
+                                            foundedRole -> foundedRole.getPrivileges()
+                                                                      .forEach(
+                                                                              this::addAnonymousAuthority)));
+
+    }
+
+    private ResponseEntity processConfig(Consumer<AuthorizationConfig> consumer,
```

Comment:
`consumer` could you please rename this argument to be more self-explained? I guess it's some kind of operation, right ?